### ISSUE TITLE
Instance scoped language

### ISSUE THREAD

**Issue by klausXR** [skaterdav85/validatorjs#401]
Is it possible to set the language on a specific instance and not globally @mikeerickson ?

We have an application which responds to user requests with dynamic languages ( one request might come in English and another one in Spanish ) and I'd like to know if I'm able to set the language on a specific Validator instance.

**Comment by mikeerickson** [comment]
@klausXR I am not sure how you have implemented `validatorjs` into your project, but the language can already be tied to a specific instance

In the following example, this would be applied to the current validation operation

```js
const currentLanguage = Validator.getDefaultLang(); // get the current language

Validator.useLang("se");  // change to Swedish

const validator = new Validator({ zip: "" }, { zip: "required" });

validator.passes();

Validator.useLang(currentLanguage); // restore default language
```

If you are looking for something else, please provide more clarification

**Comment by klausXR** [comment]
> @klausXR I am not sure how you have implemented `validatorjs` into your project, but the language can already be tied to a specific instance
> 
> In the following example, this would be applied to the current validation operation
> 
> ```js
> const currentLanguage = Validator.getDefaultLang(); // get the current language
> 
> Validator.useLang("se");  // change to Swedish
> 
> const validator = new Validator({ zip: "" }, { zip: "required" });
> 
> validator.passes();
> 
> Validator.useLang(currentLanguage); // restore default language
> ```
> 
> If you are looking for something else, please provide more clarification

Thank you for the quick reply.

The `useLang` method is a static one, so it does not influence the specific instance only ( like prototype methods do ).

I already know that I can use the restoration technique, but that is not a very good solution. If you have asynchronous validation, for example, by the time it completes, another request might have came and updated the language globally. 

What you have proposed is a solution to a certain extent, but I'm not sure I'm willing to put up with at the moment.

What I would like to accomplish is something similar to this:

```js
// Third argument is for custom error messages, this example assumes an `options` config object

const validator = new Validator({ zip: "" }, { zip: "required" }, {
    lang: "en"
})

const secondValidator = new Validator({ zip: "" }, { zip: "required" }, {
    lang: "es"
})

const thirdValidator = new Validator({ zip: "" }, { zip: "required" }, {
    lang: "fr"
})
```

Each instance has independent language and no restoration is necessary.

As there is no API to accomplish this, I'm thinking about wrapping the library ( extend the Validator class ), so that I can add the additional language argument to the constructor. Then, based on that, I can import my own locales for that language and pass them as custom messages to the `Validator` constructor. Downside is, the selected language according to the Validator will always be English. And it probably has some other downsides as well. 

I'm not really a fan of this solution either. Looking forward to your opinion on this.

**Comment by mikeerickson** [comment]
@klausXR I see what you are asking for and will put it on the feature list for a future version.

**Comment by mikeerickson** [comment]
@klausXR Just wanted to let you know, this has been implemented in upcoming 3.5 release.

```js
const validator = new Validator({ zip: "" }, { zip: "required" }, null, "se");
const message = validator.errors.first("zip")); // returns `zip måste vara ifyllt.`
```
